### PR TITLE
DNN-20086: GetPageDetails getting crashed with null object reference

### PR DESCRIPTION
**Pull request by Mohtshum**

### CODE REVIEW

File: src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs [hunk 1]
```diff
@@ -1027,14 +1027,15 @@ public IEnumerable<Url> GetPageUrls(int tabId)
 
         public PageSettings GetPageSettings(int pageId, PortalSettings requestPortalSettings = null)
         {
-            var tab = GetPageDetails(pageId);            
+            var tab = GetPageDetails(pageId);
 
-            if (!_contentVerifier.IsContentExistsForRequestedPortal(tab.PortalID, requestPortalSettings))
+            var portalSettings = requestPortalSettings ?? PortalController.Instance.GetCurrentPortalSettings();
```
Comment (tpluscode) [inline src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs]: I think that with this class we've already moved to constructor-injecting dependencies

**Review by tpluscode — CHANGES_REQUESTED** [review]
Also, please add a simple unit test to verify that `PortalController#GetCurrentPortalSettings` is called in case of null parameter

File: src/Modules/Content/Dnn.PersonaBar.Pages.Tests/PagesControllerUnitTests.cs [hunk 2]
```diff
@@ -54,32 +59,60 @@ public void ValidatePageUrlSettings_CleanNameForUrl_URLArgumentShouldBeLocalPath
...
+
+            // Arrange
+            _tabControllerMock.Setup(t => t.GetTab(It.IsAny<int>(), It.IsAny<int>())).Returns(_tab);
+            _portalControllerMock.Setup(p => p.GetCurrentPortalSettings()).Returns(_portalSettings);
+            _contentVerifierMock.Setup(c => c.IsContentExistsForRequestedPortal(It.IsAny<int>(), It.IsAny<PortalSettings>(), false)).Returns(false);
+
+            InitializePageController();
+
+            // Act
+            TestDelegate pageSettingsCall = () =>  _pagesController.GetPageSettings(tabId, null);
```
Comment (Mohtshum) [inline src/Modules/Content/Dnn.PersonaBar.Pages.Tests/PagesControllerUnitTests.cs:97]: GetPageSettings has lots of dependencies, to minimize the change set to validate the null parameter changes are made for exception throwing case.

Comment (tpluscode) [reply src/Modules/Content/Dnn.PersonaBar.Pages.Tests/PagesControllerUnitTests.cs:97]: Yeah, I see how this avoids all the setup which would be necessary to test a "happy path". It does verify what we had to verify so that's fine

File: src/Modules/Content/Dnn.PersonaBar.Pages.Tests/PagesControllerUnitTests.cs [hunk 3]
```diff
@@ -23,13 +24,16 @@ public class PagesControllerUnitTests
         Mock<IUrlRewriterUtilsWrapper> _urlRewriterUtilsWrapperMock;
         Mock<IFriendlyUrlWrapper> _friendlyUrlWrapperMock;
         Mock<IContentVerifier> _contentVerifierMock;
-
         PagesControllerImpl _pagesController;
+        Mock<IPortalController> _portalControllerMock;
+
+        TabInfo _tab = new TabInfo();
+        PortalSettings _portalSettings = new PortalSettings();
```
Comment (tpluscode) [inline src/Modules/Content/Dnn.PersonaBar.Pages.Tests/PagesControllerUnitTests.cs]: I think I's prefer to keep `TabInfo` and `PortalSettings in individual test methods

**Review by tpluscode — CHANGES_REQUESTED** [review]

**Review by tpluscode — APPROVED** [review]
